### PR TITLE
Support folding notebook cells via sticky scroll

### PR DESCRIPTION
**Pull request by Yoyokrazy**
re: #188703

### CODE REVIEW

**Comment by Yoyokrazy** [comment]
todo: 

- [ ] update testing snapshots (again) to account for changes to header content....

File: src/vs/workbench/contrib/notebook/browser/viewParts/notebookEditorStickyScroll.ts [hunk 1]
```diff
@@ -54,26 +56,46 @@ export class ToggleNotebookStickyScroll extends Action2 {
...
 	constructor(
 		public readonly element: HTMLElement,
+		public readonly foldingIcon: StickyFoldingIcon,
+		public readonly header: HTMLElement,
 		public readonly entry: OutlineEntry,
 		public readonly notebookEditor: INotebookEditor,
 	) {
 		super();
-		this._register(DOM.addDisposableListener(this.element, DOM.EventType.CLICK, () => {
+		// click the header to focus the cell
```
Comment (rebornix) [inline src/vs/workbench/contrib/notebook/browser/viewParts/notebookEditorStickyScroll.ts:65]: We might want to listen to `EventType.CLICK` and `TouchEventType.Tap` to ensure it's also working with touch screen

File: src/vs/workbench/contrib/notebook/browser/viewParts/notebookEditorStickyScroll.ts [hunk 2]
```diff
@@ -54,26 +59,56 @@ export class ToggleNotebookStickyScroll extends Action2 {
...
+		// 	this.foldingIcon.setVisible(true);
+		// }));
+		// this._register(DOM.addDisposableListener(this.element, DOM.EventType.MOUSE_OUT, () => {
+		// 	this.foldingIcon.setVisible(false);
+		// }));
+
+	}
+
+	private toggleFoldRange(currentState: CellFoldingState) {
+		const foldingController = this.notebookEditor.getContribution('workbench.notebook.foldingController') as FoldingController;
```
Comment (rebornix) [inline src/vs/workbench/contrib/notebook/browser/viewParts/notebookEditorStickyScroll.ts:92]: We can do `this.notebookEditor.getContribution<FoldingController>(FoldingController.id)

**Review by rebornix — DISMISSED** [review]
Looks good to me in overall, just one nit about how to get the editor contribution instance.

**Review by rebornix — APPROVED** [review]

**Pull request merged**